superbench/runner/runner.py

```diff
@@ -444,7 +445,20 @@ def run(self):
                     )
                     ansible_rc = sum(rc_list)
                 elif mode.name == 'torch.distributed' or mode.name == 'mpi':
-                    ansible_rc = self._run_proc(benchmark_name, mode, {'proc_rank': 0})
+                    if not mode.pattern:
+                        ansible_rc = self._run_proc(benchmark_name, mode, {'proc_rank': 0})
+                    else:
+                        hostx = self._ansible_client._host_list
+                        pattern_hostx = gen_tarffic_pattern_host_group(list(map(str, hostx)), mode)
+                        rc_list = []
+                        for host_groups in pattern_hostx:
+                            para_rc_list = Parallel(n_jobs=len(host_groups))(
+                                delayed(self._run_proc)(benchmark_name, mode, hostx=host_group, vars={
+                                    'proc_rank': 0
```

Comment:
can you pass host_list in vars and access mode.host_list in _run_proc or __get_mode_command